projects/core/src/lib/colors/color-map-selector/color-map-selector.component.ts

```diff
@@ -211,17 +211,36 @@ export class ColorMapSelectorComponent implements OnInit, OnDestroy, OnChanges {
     ): Array<ColorBreakpoint> {
         const breakpoints = new Array<ColorBreakpoint>();
 
-        const stepSize = 1 / (colorMapSteps - 1);
-        for (let frac = 0; frac <= 1; frac += stepSize) {
+        for (let i = 0; i <= colorMapSteps; i++) {
+            const frac = i / colorMapSteps;
             const value = bounds.min + frac * (bounds.max - bounds.min);
 
-            let colorMapIndex = Math.round(frac * (colorMap.length - 1));
+            const colorMapPos = frac * (colorMap.length - 1);
+            let colorMapIndex = Math.floor(colorMapPos);
+            const colorMapPosRemainder = colorMapPos - colorMapIndex;
+            let nextMapIndex = Math.min(colorMapIndex + 1, colorMap.length - 1);
+
             if (colorMapReverseColors) {
                 colorMapIndex = colorMap.length - colorMapIndex - 1;
+                nextMapIndex = Math.max(0, colorMapIndex - 1);
+            }
+
+            if (colorMapPosRemainder < Number.EPSILON) {
+                breakpoints.push(new ColorBreakpoint(value, Color.fromRgbaLike(colorMap[colorMapIndex])));
+                continue;
             }
+
             const color = Color.fromRgbaLike(colorMap[colorMapIndex]);
+            const nextColor = Color.fromRgbaLike(colorMap[nextMapIndex]);
```

Comment:
maybe add a comment here that you interpolate between two steps?

why wasn't this done here before anyway?

also: does you method always work? It seems the values can go beyond 256?

maybe this would be something that could even be checked in a unit test.

Reply:
well there was a problem when the colors are not truncated because the backend can't handle float rgba values.
The method schould always work since the frac is still the same as before just computed different. The "value" is the value of the pixel and this can be larger then 255.

Reply:
[d31466b](https://github.com/geo-engine/geoengine-ui/pull/680/commits/d31466ba837ef2fb26d0b6740bdfbc7b47c3cf37)